content/posts/07-meet-mkmr.md

```diff
@@ -0,0 +1,248 @@
+
+![Final Report](07-meet-mkmr-07.png)
+
+After merging it will delete from the cache all branches that were given as they have no
+use for us any longer.
+
+### vimr
+
+This one allows users to edit attributes of a given Merge Request by opening them in `$EDITOR`
+I think it is better shown than talked about as the explanations written to the `$EDITOR` say
+all that needs to be said.
+
+> This is the result of: vimr -n 11361
+
+```null
+# Set to 'close' to close the merge request
+# set to 'reopen' to open a closed merge request
+State: opened
+
+# Line must
+Title: community/picard: upgrade to 2.4.1
+
+# Any long multi-line string, will consider everything until the first line
+# that starts with a hashtag (#)
+Description:
+
+# Whitespace-separated list of labels, have it empty to remove all labels
+Labels:
+
+# If this is true then the source branch of the Merge Request will de deleted
+# when this Merge Request is merged, takes 'true' or 'false'
+Remove source branch when merged: True
+
+# Name of the branch the branch the commit will be land on
+Target Branch: master
+
+# If 'true' then no users can comment, 'false' otherwise
+Lock Discussion: False
+
+# If 'true' then all commits will be squashed into a single one
+Squash on merge: False
+
+# If 'true' maintainers of the repository will be able to push commits
+# into your branch, this is required for people rebasing-and-merging
+Allow Maintainers to push: True
+```
+
+### edmr
+
+Just like `vimr` it also edits attributes of Merge Requests but it only edits one at
+a time and does it from the commandline, by passing arguments to the commandline one
+can change the options listed with `edmr -l`.
+
+```sh
+$ edmr -l
+target_branch -> a single string
+state_event -> a single string
+:description -> a single string
+description -> a single string
+remove_source_branch -> boolean
+:title -> a single string
+allow_maintainer_to_push -> boolean
+squash -> boolean
+assignee_ids -> multiple integers separated by whitespace
+labels: -> one or more strings separated by whitespace
+title: -> a single string
+allow_collaboration -> boolean
+milestone_id -> a single string
+description: -> a single string
+labels -> one or more strings separated by whitespace
+discussion_locked -> boolean
+:labels -> one or more strings separated by whitespace
+assignee_id -> integer
+title -> a single string
+```
+
+the `:` before and after a value indicates that it should be prefixed or suffixed, doing
+`edmr MRNUM :title="[3.12] "` will prefix to the existing title of the Merge Request instead
+of replacing it.
+
+Basically `edmr` is the scriptable non-interactive relative of `vimr`.
+
+> Replace title with head commit: `edmr MRNUM title="$(git show --oneline --format=%s)"`
+
+#### Notes
+
+The following is a small collection of notes that I could not find a proper place to present
+to you:
+
+##### Booleans
+
+While python3 accepts `True` and `False` we have a function that accepts even more values
```

Comment:
This part seems a bit disjointed from the rest of the article, maybe just mentioning in the `mkmr` part that it takes any of these values for YES/NO would do?

Reply:
Which is why it is in notes at the end, look up

Reply:
Well yes, but it just doesn't seem relevant to the rest of the article and IMHO instead of having this at the end of the article having this as a bit of trivia in mkmr would be better